app/src/main/java/org/schabi/newpipe/player/playqueue/PlayQueueItem.java

```diff
@@ -64,20 +64,6 @@ private PlayQueueItem(@Nullable final String name, @Nullable final String url,
         this.recoveryPosition = RECOVERY_UNSET;
     }
 
-    @Override
-    public boolean equals(final Object o) {
```

Comment:
Oh my! what a bug xD
Did you check the usages of the method? Maybe someone implemented this on purpose.

Reply:
Android Studio's "Show usages" yields all of the usages of the `equals()` method on any object of any class, so that's not much useful. I tried to manually look for usages by checking out the methods in `PlayQueue` and in `Player` that use `PlayQueueItem` comparison and could not find any issue.
The only downside I can think of is that if the same video is queued multiple times in a row, the loading strategy may start to load the current and the next stream contemporarily, but since those streams point to the same url but do not compare equal, they will be loaded twice. But this only happens if two play queue items with the same url start loading contemporarily, because otherwise the caching method prevents reloading something that already was loaded. I don't think this is really a downside.